main.cpp

```diff
@@ -37,11 +38,20 @@
 #include <windows.h> /// for getting PID, for proper RNG for MinGW
 #endif
 
+volatile sig_atomic_t userExitFlag = 0;
+void catchCtrlC(int signalID) {
+  signal(SIGINT, catchCtrlC);
+  if (userExitFlag==1) exit(0); // user force quit
```

Comment:
If the user forces a quit, then we shouldn't signal a clean exit. I think the usual way is to use the signal ID that was passed in as the exit signal, i.e.
exit(signalID);
instead of
exit(0);

Reply:
Good idea; Change made. I explicitly invoke SIGTERM (x-platform) to end all threads, otherwise the main thread is still running for a bit depending on the OS.

Reply:
Yeah, SIGTERM is being raised because on Win systems the main thread keeps running after exit is called, and Archivist reports errors before the program exits. It's very unclean. SIGTERM forces all threads to be killed. The exit was indeed unnecessary after this and does not clean up the "Terminated" output as I thought on Linux systems. Oh well, it's fine.